File: Scripts/results/generated_code_results/baseline_gpt/Social Networking/SocialSoftware/FeedScreen.js
import React from 'react';
import { View, Text, StyleSheet, Image, TouchableOpacity, FlatList } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const FeedScreen = () => {
  const data = [
    {
      id: '1',
      username: 'arochinski',
      time: '2 min',
      text: 'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',
      image: '../assets/snack-icon.png',
      answers: 7,
      likes: 59,
    },
    {
      id: '2',
      username: 'aura',
      time: '1 min',
      text: 'Tempor dolor proident duis incididunt et. Cillum',
      answers: 0,
      likes: 0,
    },
  ];

  const renderItem = ({ item }) => (
    <View style={styles.postContainer}>
      <View style={styles.header}>
        <Image source={require('../assets/snack-icon.png')} style={styles.avatar} />
        <Text style={styles.username}>{item.username}</Text>
        <Text style={styles.time}>{item.time}</Text>
        <Icon name="ellipsis-horizontal" size={20} color="#FFFFFF" />
      </View>
      <Text style={styles.text}>{item.text}</Text>
      {item.image && <Image source={require('../assets/snack-icon.png')} style={styles.postImage} />}
      <View style={styles.actions}>
        <TouchableOpacity>
          <Icon name="heart-outline" size={24} color="#FFFFFF" style={styles.icon} />
        </TouchableOpacity>
        <TouchableOpacity>
          <Icon name="chatbubble-outline" size={24} color="#FFFFFF" style={styles.icon} />
        </TouchableOpacity>
        <TouchableOpacity>
          <Icon name="share-outline" size={24} color="#FFFFFF" style={styles.icon} />
        </TouchableOpacity>
      </View>
      <Text style={styles.stats}>{item.answers} answers · {item.likes} likes</Text>
    </View>
  );

  return (
    <View style={styles.container}>
      <FlatList
        data={data}
        renderItem={renderItem}
        keyExtractor={item => item.id}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#000000',
  },
  postContainer: {
    marginBottom: 20,
    paddingHorizontal: 15,
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  avatar: {
    width: 40,
    height: 40,
    borderRadius: 20,
    marginRight: 10,
  },
  username: {
    color: '#FFFFFF',
    fontWeight: 'bold',
    flex: 1,
  },
  time: {
    color: '#AAAAAA',
    marginRight: 10,
  },
  text: {
    color: '#FFFFFF',
    marginBottom: 10,
  },
  postImage: {
    width: '100%',
    height: 200,
    borderRadius: 10,
    marginBottom: 10,
  },
  actions: {
    flexDirection: 'row',
    marginBottom: 5,
  },
  icon: {
    marginRight: 15,
  },
  stats: {
    color: '#AAAAAA',
  },
});

export default FeedScreen;

File: Scripts/results/generated_code_results/Flutter/Social Networking/SocialSoftware/FeedScreen.dart
import 'package:flutter/material.dart';



class FeedScreen extends StatefulWidget {

  @override

  _FeedScreenState createState() => _FeedScreenState();

}



class _FeedScreenState extends State<FeedScreen> {

  @override

  Widget build(BuildContext context) {

    return Scaffold(

      appBar: AppBar(

        title: Image.asset('assets/images/template.png', height: 30),

        actions: [

          IconButton(

            icon: Icon(Icons.notifications),

            onPressed: () => Navigator.pushNamed(context, 'Activity'),

          ),

          IconButton(

            icon: Icon(Icons.menu),

            onPressed: () {},

          ),

        ],

      ),

      body: ListView(

        children: [

          _buildPost(

            'arochinski',

            'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',

            true,

            7,

            59,

            '2 min',

          ),

          _buildPost(

            'arochinski',

            'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',

            false,

            7,

            59,

            '2 min',

          ),

          _buildPost(

            'aura',

            'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',

            false,

            0,

            0,

            '1min',

          ),

        ],

      ),

      bottomNavigationBar: BottomNavigationBar(

        type: BottomNavigationBarType.fixed,

        items: [

          BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),

          BottomNavigationBarItem(icon: Icon(Icons.search), label: 'Search'),

          BottomNavigationBarItem(icon: Icon(Icons.add_box), label: 'Create'),

          BottomNavigationBarItem(icon: Icon(Icons.favorite), label: 'Likes'),

          BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Profile'),

        ],

        onTap: (index) {

          switch (index) {

            case 1:

              Navigator.pushNamed(context, 'Search');

              break;

            case 4:

              Navigator.pushNamed(context, 'Profile');

              break;

          }

        },

      ),

    );

  }



  Widget _buildPost(String username, String content, bool hasImage, int answers, int likes, String timeAgo) {

    return Column(

      crossAxisAlignment: CrossAxisAlignment.start,

      children: [

        ListTile(

          leading: GestureDetector(

            onTap: () => Navigator.pushNamed(context, 'UserProfile'),

            child: CircleAvatar(

              backgroundImage: AssetImage('assets/images/template.png'),

            ),

          ),

          title: Text(username),

          subtitle: Text(timeAgo),

          trailing: IconButton(

            icon: Icon(Icons.more_horiz),

            onPressed: () {},

          ),

        ),

        Padding(

          padding: const EdgeInsets.symmetric(horizontal: 16.0),

          child: Text(content),

        ),

        if (hasImage)

          Padding(

            padding: const EdgeInsets.all(16.0),

            child: Image.asset('assets/images/template.png'),

          ),

        Row(

          mainAxisAlignment: MainAxisAlignment.spaceAround,

          children: [

            IconButton(

              icon: Icon(Icons.chat_bubble_outline),

              onPressed: () {},

            ),

            Text('$answers'),

            IconButton(

              icon: Icon(Icons.favorite_border),

              onPressed: () {},

            ),

            Text('$likes'),

            IconButton(

              icon: Icon(Icons.share),

              onPressed: () {},

            ),

          ],

        ),

        Divider(),

      ],

    );

  }

}